include/zephyr/drivers/rtc.h

```diff
@@ -527,6 +529,22 @@ static inline struct tm *rtc_time_to_tm(struct rtc_time *timeptr)
 	return (struct tm *)timeptr;
 }
 
+/**
+ * @brief Determine required calibration to 1 Hertz from frequency.
+ *
+ * @param frequency Frequency of the RTC in nano Hertz
+ *
+ * @return The required calibration in parts per billion
+ */
+static inline int32_t rtc_calibration_from_frequency(uint32_t frequency)
+{
+	if (frequency == 0) {
+		return INT32_MAX;
+	}
+
+	return (int32_t)((1000000000000000000 / frequency) - 1000000000);
```

Comment:
Don't you need to tell the compiler that `1000000000000000000` is 64 bit?

Reply:
In this case, the compiler will chose a type which can store 1000000000000000000, and promote any other variable to the same size while doing arithmetic operations, so every variable will become an int64_t implicitly :)

Reply:
I'd prefer using the `LL` suffix.

Reply:
updated to specify `LL` :)